### ISSUE TITLE
Crash in wait, on  macOS

### ISSUE THREAD

**Issue by gaborcsardi** [r-lib/processx#138]
E.g. https://travis-ci.org/r-lib/processx/jobs/406825805

Seems like a race condition when a finalizer or a SIGCHILD handler (?) is called while waiting.

**Comment by gaborcsardi** [comment]
This helps: https://travis-ci.org/r-lib/processx/jobs/406887924

It seems that the sigchld arrives in the middle of the polling, which should be OK in general, but seemingly it is not?

**Comment by gaborcsardi** [comment]
Some modes of failure:
```
sigchld remove  8525
[1] "call to wait"
wait
handle active
setup sigchld
block sigchld after setting up
polling
sigchld remove  8526
polling
[1] "1"
 *** caught segfault ***
address 0xffffffff8b023d68, cause 'memory not mapped'
```

```
sigchld remove  6727
sigchld remove  6728
sigchld remove  6729
sigchld remove  6730
 *** caught segfault ***
address 0xffffffffb2ae0a00, cause 'memory not mapped'
```

The first  one seems to  be more  frequent...

**Comment by gaborcsardi** [comment]
This was the first time it happened: https://travis-ci.org/r-lib/processx/jobs/389200807

**Comment by gaborcsardi** [comment]
Fixed by  #142.
